addons/udes_stock/models/stock_move_line.py

```diff
@@ -1119,10 +1119,12 @@ def _prevent_write_after_done(self):
 
         # When a moveline is updated through desktop only one "done" move line is passed into self even when there are multiple
         # Get all the done move lines on the picking if there is only one "done" move line
-        if len(done_lines) == 1 and self.mapped("picking_id"):
-            done_lines = self.search(
-                [("picking_id", "in", self.mapped("picking_id.id")), ("state", "=", "done")]
-            )
+        if len(done_lines) == 1:
+            picking_ids = self.mapped("picking_id")
+            if picking_ids:
+                done_lines = self.search(
+                    [("picking_id", "in", self.mapped("picking_id.id")), ("state", "=", "done")], order="id"
```

Comment:
Mistake on my comment sorry, should be `("picking_id", "in", picking_ids)` - as we already have the picking_ids from line 1123.